packages/odyssey-react-mui/src/labs/TopNav.tsx

```diff
@@ -0,0 +1,459 @@
+    borderRadius: 0,
+    outlineWidth: odysseyDesignTokens.FocusOutlineWidthMain,
+    backgroundColor: !isDisabled ? odysseyDesignTokens.HueNeutral50 : "inherit",
+  },
+}));
+
+const NavItemContentClickContainer = styled("div", {
+  shouldForwardProp: (prop) =>
+    prop !== "odysseyDesignTokens" && prop !== "isDisabled",
+})<{
+  odysseyDesignTokens: DesignTokens;
+  isDisabled?: boolean;
+}>(({ odysseyDesignTokens, isDisabled }) => ({
+  display: "flex",
+  alignItems: "center",
+  width: "100%",
+  padding: `${odysseyDesignTokens.Spacing2} ${odysseyDesignTokens.Spacing4}`,
+  color: `${isDisabled ? odysseyDesignTokens.TypographyColorDisabled : odysseyDesignTokens.TypographyColorHeading} !important`,
+  "&:focus-visible": {
+    borderRadius: 0,
+    outlineColor: odysseyDesignTokens.FocusOutlineColorPrimary,
+    outlineStyle: odysseyDesignTokens.FocusOutlineStyle,
+    outlineWidth: odysseyDesignTokens.FocusOutlineWidthMain,
+    backgroundColor: odysseyDesignTokens.HueNeutral50,
+    textDecoration: "none",
+  },
+}));
+
+const TopNavItemContent = ({
+  id,
+  label,
+  href,
+  target,
+  onClick,
+  isDisabled,
+}: TopNavLinkItem) => {
+  const odysseyDesignTokens = useOdysseyDesignTokens();
+
+  const topNavItemContentKeyHandler = useCallback(
+    (event: KeyboardEvent<HTMLDivElement>) => {
+      if (event?.key === "Enter") {
+        event.preventDefault();
+        event.stopPropagation();
+        onClick?.();
+      }
+    },
+    [onClick],
+  );
+
+  const TopNavItemContent = useMemo(() => {
+    return (
+      <TopNavListItemContainer
+        id={id}
+        key={id}
+        aria-disabled={isDisabled}
+        isDisabled={isDisabled}
+        odysseyDesignTokens={odysseyDesignTokens}
+      >
+        {
+          // Use Link for nav items with links and div for disabled or non-link items
+          isDisabled ? (
+            <NavItemContentClickContainer
+              odysseyDesignTokens={odysseyDesignTokens}
+            >
+              <TopNavItemLabelContainer
+                odysseyDesignTokens={odysseyDesignTokens}
+              >
+                {label}
+              </TopNavItemLabelContainer>
+            </NavItemContentClickContainer>
+          ) : !href ? (
+            <NavItemContentClickContainer
+              odysseyDesignTokens={odysseyDesignTokens}
+              role="button"
+              tabIndex={0}
+              onClick={onClick}
+              onKeyDown={topNavItemContentKeyHandler}
+            >
+              <TopNavItemLabelContainer
+                odysseyDesignTokens={odysseyDesignTokens}
+              >
+                {label}
+              </TopNavItemLabelContainer>
+            </NavItemContentClickContainer>
+          ) : (
+            <Link href={href} target={target} onClick={onClick}>
+              <TopNavItemLabelContainer
+                odysseyDesignTokens={odysseyDesignTokens}
+              >
+                {label}
+              </TopNavItemLabelContainer>
```

Comment:
I'm wondering if we want to add some `role` attributes in here. Some of these `div`s are gonna be hard to select without the correct roles define, but looking through here and not seeing it visually, I can't quite say.

Reply:
hmm, not sure if any of these are clickable/selectable items. We will have links and buttons in the nav and the `li -> div` container already has a button role when it is not a link